js/core/utils/data.js

```diff
@@ -181,7 +181,9 @@ export const compileSetter = function(expr) {
     };
 };
 
-export const toComparable = function(value, caseSensitive) {
+export const toComparable = function(value, caseSensitive, options) {
+    options = options || {};
```

Comment:
Let's not override function arguments.

There are two variants:

**add a new variable e.g ```comparableOptions```**
```typescript
const comparableOptions = options || {};
```
And use this variable in the function code.
Btw, if we choose this way, let's remove ```?``` after ```options``` -> ```comparableOptions```
```typescript
// we don't need the ? symbol after the comparableOptions because it's always defined
if(comparableOptions.collateOptions?.sensitivity === 'base')
```

**add the ```?``` symbol everywhere after ```options```**
Because ```options``` may be null / undefined we can handle this just add the ```?``` after it everywhere :)

Reply:
Better to just use default value
```
function(value, caseSensitive, options = {}) {
```